lib/service/service.go

```diff
@@ -57,6 +59,10 @@ func (svc *LndhubService) GenerateToken(ctx context.Context, login, password, in
 		}
 	}
 
+	if user.Deactivated {
+		return "", "", fmt.Errorf(responses.AccountDeactivatedError.Message)
+	}
+
```

Comment:
can we do this check (and potential other checks in an middleware that runs before each secured endpoint. 
This has the advantage that it is checked on each request and also that this `GenerateToken` function does not have the sideeffect of checking some user details

Reply:
That would introduce an extra database operation for every request but it would be possible.

Reply:
A lookup of the user is cheap - and I guess in many cases we do this already (though I guess we can not reuse the loaded user)

Reply:
We almost never do this in the request handling. We always use the user id from the token and never perform the database operation to fetch the user.